File: coding_agent/backend/workspace/test_solution.py
from solution import calculate_factorial

def test_calculate_factorial_happy_path():
    assert calculate_factorial(5) == 120
    assert calculate_factorial(3) == 6
    assert calculate_factorial(1) == 1
    assert calculate_factorial(4) == 24

def test_calculate_factorial_edge_cases():
    assert calculate_factorial(0) == 1
    assert calculate_factorial(None) == "Error: Input must be a non-negative integer"
    assert calculate_factorial(-1) == "Error: Input must be a non-negative integer"

def test_calculate_factorial_boundary_values():
    assert calculate_factorial(2) == 2
    assert calculate_factorial(10) == 3628800

def test_calculate_factorial_large_number():
    assert calculate_factorial(20) == 2432902008176640000

def test_calculate_factorial_non_integer_input():
    assert calculate_factorial(5.5) == "Error: Input must be a non-negative integer"
    assert calculate_factorial("five") == "Error: Input must be a non-negative integer"

File: coding_agent/backend/workspace/test_solution.ts
import { solution } from './solution';

describe('Factorial Function Tests', () => {
    test('should return 1 for factorial of 0', () => {
        expect(solution(0)).toBe(1);
    });

    test('should return 1 for factorial of 1', () => {
        expect(solution(1)).toBe(1);
    });

    test('should return 120 for factorial of 5', () => {
        expect(solution(5)).toBe(120);
    });

    test('should return 720 for factorial of 6', () => {
        expect(solution(6)).toBe(720);
    });

    test('should return 40320 for factorial of 8', () => {
        expect(solution(8)).toBe(40320);
    });

    test('should throw an error for negative input', () => {
        expect(() => solution(-1)).toThrow('Input must be a non-negative integer');
    });

    test('should return 1 for factorial of 2 (boundary case)', () => {
        expect(solution(2)).toBe(2);
    });

    test('should return 3628800 for factorial of 10 (large input)', () => {
        expect(solution(10)).toBe(3628800);
    });
});